onnxruntime/python/session.py

```diff
@@ -194,18 +194,31 @@ class IOBinding:
     def __init__(self, session):
         self._iobinding = C.SessionIOBinding(session._sess)
 
-    def bind_input(self, name, device_type, device_id, element_type, shape, buffer_ptr):
-        self._iobinding.bind_input(name,
+    def bind_input(self, name, device_type=None, device_id=None, element_type=None, shape=None, buffer_ptr=None):
+        if device_type is None:
```

Comment:
What's the use case for binding an input where you don't provide the actual input?

Reply:
reverted